### ISSUE TITLE
tiledb cloud quickstart example throws error on WSL/Ubuntu

### ISSUE THREAD

**Issue by bkmartinjr** [TileDB-Inc/TileDB-Py#241]
Trying to run the "quickstart" cloud array access example succeeds on an EC2 Linux instance, but fails on Windows/WSL/Ubuntu (tiledb works great on Win10/WSL).  Test program is below.

Running on Win10/WSL/Ubuntu, I see an error:
```
# python cloud.py
Traceback (most recent call last):
  File "cloud.py", line 17, in <module>
    fetch()
  File "cloud.py", line 12, in fetch
    with tiledb.DenseArray("tiledb://TileDB-Inc/quickstart_dense", ctx=tiledb.Ctx(config)) as A:
  File "tiledb/libtiledb.pyx", line 3560, in tiledb.libtiledb.DenseArrayImpl.__init__
  File "tiledb/libtiledb.pyx", line 3038, in tiledb.libtiledb.Array.__init__
  File "tiledb/libtiledb.pyx", line 387, in tiledb.libtiledb._raise_ctx_err
  File "tiledb/libtiledb.pyx", line 372, in tiledb.libtiledb._raise_tiledb_error
tiledb.libtiledb.TileDBError: [TileDB::REST] Error: Error in libcurl GET operation: libcurl error message 'error setting certificate verify locations:
  CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: /etc/ssl/certs/ca-certificates.crt'; HTTP code 0; server response was empty.
```

running on the EC2 instance, it works (edited for brevity):
```
$ python cloud.py
- Array type: dense
- Cell order: row-major
- Tile order: row-major
- Capacity: 10000
- Coordinates compressor: ZSTD
- Coordinates compression level: -1

=== Domain ===
- Dimensions type: INT32
...
```

Code (access token removed):

```
import tiledb
import tiledb.cloud
import numpy

_token = "a secret"


def fetch():
    config = tiledb.Config()
    config["rest.token"] = _token

    with tiledb.DenseArray("tiledb://TileDB-Inc/quickstart_dense", ctx=tiledb.Ctx(config)) as A:
        A.dump()


if __name__ == '__main__':
    fetch()
```

**Comment by mckinsel** [comment]
FWIW, I've had similar issues when running tiledb.cloud anywhere other than an EC2 instance. Here are the things I tried from my local machine using this script

```python
import tiledb, tiledb.cloud
import pandas

print(tiledb.__version__)
print(tiledb.libtiledb.version())

tiledb.cloud.login(token="...")

array_name = "tiledb://TileDB-Inc/quickstart_sparse"

with tiledb.SparseArray(array_name, 'r', ctx=tiledb.cloud.Ctx()) as A:
    data = A[:]
    print(pandas.DataFrame(data))
```

I first tried in an ubuntu:18.04 docker container:
```
$ pip install tiledb-cloud
$ python test.py
0.5.0
(1, 7, 0)
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    with tiledb.SparseArray(array_name, 'r', ctx=tiledb.cloud.Ctx()) as A:
  File "tiledb/libtiledb.pyx", line 4101, in tiledb.libtiledb.SparseArrayImpl.__init__
  File "tiledb/libtiledb.pyx", line 3038, in tiledb.libtiledb.Array.__init__
  File "tiledb/libtiledb.pyx", line 387, in tiledb.libtiledb._raise_ctx_err
  File "tiledb/libtiledb.pyx", line 372, in tiledb.libtiledb._raise_tiledb_error
tiledb.libtiledb.TileDBError: [TileDB::REST] Error: Error in libcurl GET operation: libcurl error message 'error setting certificate verify locations:
  CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: /etc/ssl/certs/ca-certificates.crt'; HTTP code 0; server response was empty.

$ apt list --installed | grep -P "(curl|cert)"
ca-certificates/bionic,bionic-updates,now 20180409 all [installed]
ca-certificates-java/bionic,now 20170930ubuntu1 all [installed,upgradable to: 20180516ubuntu1~18.04.1]
curl/bionic-updates,bionic-security,now 7.58.0-2ubuntu3.8 amd64 [installed]
libcurl3-gnutls/now 7.58.0-2ubuntu3.1 amd64 [installed,upgradable to: 7.58.0-2ubuntu3.8]
libcurl4/bionic-updates,bionic-security,now 7.58.0-2ubuntu3.8 amd64 [installed]
libcurl4-openssl-dev/bionic-updates,bionic-security,now 7.58.0-2ubuntu3.8 amd64 [installed]
python-certifi/bionic,now 2018.1.18-2 all [installed,automatic]
ssl-cert/bionic,now 1.0.39 all [installed,automatic]

$ wc /etc/ssl/certs/ca-certificates.crt
  3408   3674 207436 /etc/ssl/certs/ca-certificates.crt
```

Then I did the same thing in a fresh centos:8 container:

```
# python3 test.py
0.5.0
(1, 7, 0)
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    with tiledb.SparseArray(array_name, 'r', ctx=tiledb.cloud.Ctx()) as A:
  File "tiledb/libtiledb.pyx", line 4101, in tiledb.libtiledb.SparseArrayImpl.__init__
  File "tiledb/libtiledb.pyx", line 3038, in tiledb.libtiledb.Array.__init__
  File "tiledb/libtiledb.pyx", line 387, in tiledb.libtiledb._raise_ctx_err
  File "tiledb/libtiledb.pyx", line 372, in tiledb.libtiledb._raise_tiledb_error
tiledb.libtiledb.TileDBError: [TileDB::REST] Error: Error in libcurl GET operation: libcurl error message 'SSL certificate problem: unable to get local issuer certificate'; HTTP code 0; server response was empty.

# yum list installed | grep -P "(curl|cert)"
ca-certificates.noarch             2018.2.24-6.el8                    @System
curl.x86_64                        7.61.1-8.el8                       @System
libcurl-minimal.x86_64             7.61.1-8.el8                       @System

# wc /etc/pki/tls/certs/ca-bundle.crt
  3646   4374 210096 /etc/pki/tls/certs/ca-bundle.crt
```

I saw some pretty recent libcurl config changes in the TileDB repo, so I thought I would try building from source and testing that back in ubuntu 18.04:
```
$ git clone https://github.com/TileDB-Inc/TileDB$ cd TileDB
$ git checkout tags/1.7.0$ mkdir build; cd build$ ../bootstrap --enable-s3
$ make -j3
$ make check
```
That gave me quite a few libcurl linking errors, so I've probably got something screwed up locally.
Repeating that in the centos:8 container worked better, but make check still failed with lots of these:
```
1: /TileDB/test/src/unit-capi-consolidation.cc:1684: FAILED:
1:   CHECK( a[i] == c_a[i] )
1: with expansion:
1:   -2147483648 == 166
```

**Comment by ihnorton** [comment]
Thanks @bkmartinjr @mckinsel, we've identified the problem, and it is fixed by the following PR:

https://github.com/TileDB-Inc/TileDB/pull/1452

I am working on a snapshot release of TileDB-Py with the fix.

**Comment by ihnorton** [comment]
I've posted snapshot builds including the fix:

https://tiledb-isaiah-public.s3.amazonaws.com/0.5.2-ca-hotfix/tiledb-0.5.2-cp27-cp27mu-manylinux1_x86_64.whl
https://tiledb-isaiah-public.s3.amazonaws.com/0.5.2-ca-hotfix/tiledb-0.5.2-cp35-cp35m-manylinux1_x86_64.whl
https://tiledb-isaiah-public.s3.amazonaws.com/0.5.2-ca-hotfix/tiledb-0.5.2-cp36-cp36m-manylinux1_x86_64.whl
https://tiledb-isaiah-public.s3.amazonaws.com/0.5.2-ca-hotfix/tiledb-0.5.2-cp37-cp37m-manylinux1_x86_64.whl

These can be installed directly w/ pip, for the matching python version:
```
pip3 install https://tiledb-isaiah-public.s3.amazonaws.com/0.5.2-ca-hotfix/tiledb-0.5.2-cp36-cp36m-manylinux1_x86_64.whl
```

(built from https://github.com/TileDB-Inc/TileDB/commit/442257d9cf997c22dc06647e2d645476953a23e8 and https://github.com/TileDB-Inc/TileDB-Py/commit/ec16d1e50777428f9586d160b631ae46439e766e)

I've tested the Python 3.6 wheel linked above on Ubuntu 18.04 in docker and EC2, and on Ubuntu 18.04 in Windows 10/WSL1.

Will leave this issue open and update when the fix is released on PyPI/conda.

**Comment by ihnorton** [comment]
@mckinsel the curl link errors should be fixed on the most recent release tag, `1.7.1` (https://github.com/TileDB-Inc/TileDB/pull/1434). I just tested a clean build on Ubuntu 18.04 with:
```
git clone https://github.com/TileDB-Inc/TileDB
cd TileDB
git checkout 1.7.1
mkdir build; cd build
../bootstrap --enable-s3 --enable-serialization
make -jN
```

**Comment by ihnorton** [comment]
This fix is released in TileDB-Py 0.5.3 on [PyPI](https://pypi.org/project/tiledb/) and [anaconda](https://anaconda.org/conda-forge/tiledb-py/files). Please comment/reopen if needed.
